internal/controlplane/handlers_authz.go

```diff
@@ -632,23 +627,134 @@ func (s *Server) UpdateRole(ctx context.Context, req *minder.UpdateRoleRequest)
 		return nil, util.UserVisibleError(codes.InvalidArgument, err.Error())
 	}
 
-	// We may be given a human-readable identifier which can vary over time. Resolve
-	// it to an IDP-specific stable identifier so that we can support subject renames.
-	identity := &auth.Identity{
-		Provider:  nil,
-		UserID:    sub,
-		HumanName: sub,
+	// Validate the subject and email - decide if it's about updating an invitation or a role assignment
+	if sub == "" && email != "" {
```

Comment:
IMO, this feels like it should be "if specified user is not in current OpenFGA roles, then look for invitations..."

Reply:
But I'm okay leaving it for now.